wal/wal.go

```diff
@@ -228,29 +228,25 @@ func NewSize(logger log.Logger, reg prometheus.Registerer, dir string, segmentSi
 	}
 
 	_, j, err := w.Segments()
+	// Index of the Segment we want to open and write to.
+	writeSegmentIndex := 0
 	if err != nil {
 		return nil, errors.Wrap(err, "get segment range")
 	}
-	// Fresh dir, no segments yet.
-	if j == -1 {
-		segment, err := CreateSegment(w.dir, 0)
-		if err != nil {
-			return nil, err
-		}
+	// If some segments already exist create one with a higher index than the last segment.
```

Comment:
Wouldn't it still be j + 1 in the j == -1 case?

Reply:
`Segments()` returns -1 for first and last if the WAL dir exists but there are no segment files present. So in that case we won't change `writeSegmentIndex` and well create the segment `0`.